d2cli/main.go

```diff
@@ -749,26 +737,38 @@ func renderPDF(ctx context.Context, ms *xmain.State, plugin d2plugin.Plugin, opt
 		if err != nil {
 			return svg, err
 		}
-		err = doc.AddPDFPage(pngImg, currBoardPath, opts.ThemeID, rootFill, diagram.Shapes, int64(opts.Pad), viewboxX, viewboxY, pageMap)
+		err = doc.AddPDFPage(pngImg, boardPath, opts.ThemeID, rootFill, diagram.Shapes, int64(opts.Pad), viewboxX, viewboxY, pageMap)
 		if err != nil {
 			return svg, err
 		}
 	}
 
 	for _, dl := range diagram.Layers {
-		_, err := renderPDF(ctx, ms, plugin, opts, "", page, ruler, dl, doc, currBoardPath, LAYERS, pageMap)
+		path := append(boardPath, pdf.BoardTitle{
+			Name:    dl.Name,
+			BoardID: strings.Join([]string{boardPath[len(boardPath)-1].BoardID, LAYERS, dl.Name}, "."),
+		})
+		_, err := renderPDF(ctx, ms, plugin, opts, "", page, ruler, dl, doc, path, pageMap)
 		if err != nil {
 			return nil, err
 		}
 	}
 	for _, dl := range diagram.Scenarios {
-		_, err := renderPDF(ctx, ms, plugin, opts, "", page, ruler, dl, doc, currBoardPath, SCENARIOS, pageMap)
+		path := append(boardPath, pdf.BoardTitle{
+			Name:    dl.Name,
+			BoardID: strings.Join([]string{boardPath[len(boardPath)-1].BoardID, SCENARIOS, dl.Name}, "."),
+		})
+		_, err := renderPDF(ctx, ms, plugin, opts, "", page, ruler, dl, doc, path, pageMap)
 		if err != nil {
 			return nil, err
 		}
 	}
 	for _, dl := range diagram.Steps {
-		_, err := renderPDF(ctx, ms, plugin, opts, "", page, ruler, dl, doc, currBoardPath, STEPS, pageMap)
+		path := append(boardPath, pdf.BoardTitle{
+			Name:    dl.Name,
+			BoardID: strings.Join([]string{boardPath[len(boardPath)-1].BoardID, STEPS, dl.Name}, "."),
```

Comment:
this still smells. how about boardPath is a slice of `BoardTitle`, and BoardTitle has
- Name
- Type

and that's it. the boardID can be derived when it's used.

Reply:
It can, given that we have special rules for `root` since its ID is just `root` and not `type.root`.
Then, I wonder if it makes sense for `pdf.BoardTilte` to know the logic of `BoardIDs` built by `buildBoardIDToIndex`. Assuming `pdf` is a separate package, it doesn't make much to have this heavily coupled assumption about the IDs in it.

Reply:
oh i see. okay, don't have to do that then. can u get rid of boardtype in ppt funcs too though